js/app.js

```diff
@@ -11,11 +11,53 @@ Calculator.prototype.getHistoryAsString = function() {
     return this.history.join('\n');
 }
 
-Calculator.prototype.add = function(num1, num2) {
-    // 1. zamień wartości przekazane przez parametr na typ number
-    // 2. sprawdź czy są one poprawne
-    // 3. jeśli tak to wykonaj działanie i zapisz jego resultat
-    // 4. dodaj do historii operacji to działanie w fomie: 1 + 1 = 2
+Calculator.prototype.validateNumbers = function(num1, num2) {
+    const number1 = Number(num1);
+    const number2 = Number(num2);
+    if (!isNaN(number1) && !isNaN(number2)) {
+        return [number1, number2];
+    }
+    
+};
+
+Calculator.prototype.performOperation = function(num1, num2, action) {
+    const numbers = this.validateNumbers(num1, num2);
+    if (!numbers) return;
+    const [number1, number2] = numbers;
+
+    let result;
+    switch (action) {
+        case '+':
+            result = number1 + number2;
+            console.log(result);
+            break;
```

Comment:
Jest to ok, można też zrobić jeszcze jeden krok do przodu. Zobacz ten artykuł:
https://devmentor.pl/b/mniej-instrukcji-warunkowych